website/pages/api-docs/volumes.mdx

```diff
@@ -0,0 +1,235 @@
+| ---------------- | --------------------------- |
+| `YES`            | `namespace:csi-list-volume` |
+
+### Parameters
+
+- `type` `(string: "")`- Specifies a string to filter volumes based on
+  the volume type. Currently supports only `csi`. This is specified as
+  a query string parameter.
+
+- `node_id` `(string: "")` - Specifies a string to filter volumes
+  based on an Node ID prefix. Because the value is decoded to bytes,
+  the prefix must have an even number of hexadecimal characters
+  (0-9a-f). This is specified as a query string parameter.
+
+- `plugin_id` `(string: "")` - Specifies a string to filter volumes
+  based on a plugin ID prefix. Because the value is decoded to bytes,
+  the prefix must have an even number of hexadecimal characters
+  (0-9a-f). This is specified as a query string parameter.
+
+### Sample Request
+
+```shell
+$ curl \
+    https://localhost:4646/v1/volumes
+```
+
+```shell
+$ curl \
+    https://localhost:4646/v1/volumes?type=csi&node_id=foo&plugin_id=plugin-id1
+```
+
+### Sample Response
+
+```json
+[
+  {
+    "ID": "volume-id1",
+    "ExternalID": "volume-id1",
+    "Namespace": "default",
+    "Name": "volume id1",
+    "Topologies": [
+      {
+        "foo": "bar"
+      }
+    ],
+    "AccessMode": "multi-node-single-writer",
+    "AttachmentMode": "file-system",
+    "Schedulable": true,
+    "PluginID": "plugin-id1",
+    "Provider": "ebs",
+    "ControllerRequired": true,
+    "ControllersHealthy": 3,
+    "ControllersExpected": 3,
+    "NodesHealthy": 15,
+    "NodesExpected": 18,
+    "ResourceExhausted": 0,
+    "CreateIndex": 42,
+    "ModifyIndex": 64,
+  }
+]
+```
+
+## Read Volume
+
+This endpoint reads information about a specific volume.
+
+| Method | Path                        | Produces           |
+| ------ | --------------------------- | ------------------ |
+| `GET`  | `/v1/volume/csi/:volume_id` | `application/json` |
+
+The table below shows this endpoint's support for
+[blocking queries](/api-docs#blocking-queries) and
+[required ACLs](/api-docs#acls).
+
+| Blocking Queries | ACL Required         |
+| ---------------- | -------------------- |
+| `YES`            | `namespace:csi-read-volume` |
+
+### Parameters
+
+- `:volume_id` `(string: <required>)`- Specifies the ID of the
+  volume. This must be the full ID. This is specified as part of the
+  path.
+
+### Sample Request
+
+```shell
+$ curl \
+    https://localhost:4646/v1/volume/csi/volume-id1
+```
+
+### Sample Response
+
+```json
+{
+  "ID": "volume-id1",
+  "Name": "volume id1",
+  "Namespace": "default",
+  "ExternalID": "volume-id1",
+  "Topologies": [
+    {"foo": "bar"}
+  ],
+  "AccessMode": "multi-node-single-writer",
+  "AttachmentMode": "file-system",
+  "Allocations": [
+    {
+      "ID": "a8198d79-cfdb-6593-a999-1e9adabcba2e",
+      "EvalID": "5456bd7a-9fc0-c0dd-6131-cbee77f57577",
+      "Name": "example.cache[0]",
+      "NodeID": "fb2170a8-257d-3c64-b14d-bc06cc94e34c",
+      [...] AllocListStub [...]
```

Comment:
This looks like it's from the RFC, should we include the full text of the `AllocListStub` here?